scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,203 @@
+package org.oppia.android.scripts
+
+import java.io.File
+import java.io.FileInputStream
+import org.oppia.android.app.model.FilenameChecks
+import org.oppia.android.app.model.FilenameCheck
+import org.oppia.android.app.model.FileContentChecks
+import org.oppia.android.app.model.FileContentCheck
+
+class RegexPatternValidationCheck {
+  companion object {
+    @JvmStatic
+    fun main(vararg args: String) {
+
+      val repoPath = args[0] + "/"
+
+      val allowedDirectories: MutableList<String> = ArrayList()
+
+      for (i in 1 until args.size) {
+        allowedDirectories.add(args[i])
+      }
+
+      val searchFiles = collectSearchFiles(repoPath, allowedDirectories)
+
+      var scriptFailedFlag = false
+
+      getFilenameChecks().forEach {
+        if (checkProhibitedFileNamePattern(
+            repoPath = repoPath,
+            searchFiles = searchFiles,
+            prohibitedFilenameRegexString = it.getProhibitedFilenameRegex(),
+            errorToShow = it.getFailureMessage()
+          )) {
+          scriptFailedFlag = true
+        }
+      }
+
+      getFileContentChecks().forEach {
+        if (checkProhibitedContent(
+            repoPath = repoPath,
+            searchFiles = searchFiles,
+            fileNameRegexString = it.getFilenameRegex(),
+            prohibitedContentRegexString = it.getProhibitedContentRegex(),
+            errorToShow = it.getFailureMessage()
+          )) {
+          scriptFailedFlag = true
+        }
+      }
+
+      if (scriptFailedFlag) {
+        throw Exception("REGEX PATTERN CHECKS FAILED")
+      } else {
+        println("REGEX PATTERN CHECKS PASSED")
+      }
+    }
+
+    fun getFilenameChecks(): List<FilenameCheck> {
```

Comment:
add kdocs

Reply:
Done

Reply:
Nit on naming: ``get`` usually refers to methods that are trivially retrieving data. Instead, consider: ``retrieveFilenameChecks`` since that more explicitly implies work is being done to fetch the data being returned & to not be confused with a simple getter.

Reply:
Thanks, done.

Reply:
@anandwana001 Can we resolve this?